### PR TITLE
[feat]: [CDS-79475]: Added GitDetails in responseDTOs for Environments.

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes:
Change: Have added entityGitDetails and gitX related fields in Env entity and Also added query params to get API.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52991)

### CODE REVIEW

**Comment by senjucanon2** [comment]
trigger messagecheck

**Comment by senjucanon2** [comment]
trigger uts

**Comment by senjucanon2** [comment]
trigger uts

**Review by hingerabhinav — DISMISSED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52991)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Comment by hingerabhinav** [comment]
@senjucanon2 desc says added query params to create. do you mean get?

**Comment by senjucanon2** [comment]
trigger uts

**Comment by senjucanon2** [comment]
trigger ut5

**Comment by senjucanon2** [comment]
> @senjucanon2 desc says added query params to create. do you mean get?

sorry typo error, it was for get flow

**Review by hingerabhinav — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52991)* status: 6 file(s) to review. 1 of 2 approvals obtained.

**Comment by yogesh-chauhan** [comment]
*[`960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/BUILD.bazel` line 22 at r5](https://reviewable.io/reviews/harness/harness-core/52991#-NemTm0pEiXYxPAzKO6A:-NemTm0q1wnpGS1mGT4C:b-7su0p7) ([raw file](https://github.com/harness/harness-core/blob/3d767d9b28ff264b7d2beef19fbf073ac0368b68/960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/BUILD.bazel#L22)):*
> ```Starlark
>         "//:lombok",
>         "//960-ng-core-beans/src/main/java/io/harness/ng/core/environment/beans:module",
>         "//960-ng-core-beans/src/main/java/io/harness/ng/core/template:module",
> ```

why do we need to import ng/core/template here ?

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52991)* status: 6 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/BUILD.bazel` line 22 at r5](https://reviewable.io/reviews/harness/harness-core/52991#-NemTm0pEiXYxPAzKO6A:-NemUSN5CrVrp0jIFKWn:bhro91a) ([raw file](https://github.com/harness/harness-core/blob/3d767d9b28ff264b7d2beef19fbf073ac0368b68/960-ng-core-beans/src/main/java/io/harness/ng/core/environment/dto/BUILD.bazel#L22)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

why do we need to import ng/core/template here ?
</blockquote></details>

We need CacheResponseMetadataDTO in the response and for this to visible in the EnvironmentResponseDTO, this was imported.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r3, 2 of 2 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52991)* status: :shipit: complete! ✓

**Review by tchaurasiya — APPROVED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java [hunk 1]
```diff
@@ -237,7 +239,13 @@ public class EnvironmentResourceV2 {
       @Parameter(description = NGCommonEntityConstants.PROJECT_PARAM_MESSAGE) @QueryParam(
           NGCommonEntityConstants.PROJECT_KEY) @ProjectIdentifier String projectIdentifier,
       @Parameter(description = "Specify whether Environment is deleted or not") @QueryParam(
-          NGCommonEntityConstants.DELETED_KEY) @DefaultValue("false") boolean deleted) {
+          NGCommonEntityConstants.DELETED_KEY) @DefaultValue("false") boolean deleted,
+      @Parameter(description = "This contains details of Git Entity like Git Branch info",
+          hidden = true) @BeanParam GitEntityFindInfoDTO gitEntityBasicInfo,
+      @Parameter(description = "Specifies whether to load the entity from cache", hidden = true) @HeaderParam(
+          "Load-From-Cache") @DefaultValue("false") String loadFromCache,
```
Comment (tchaurasiya) [inline 120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java:246]: why this parameter is string??

**Comment by senjucanon2** [comment]
trigger review

File: 120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java (hunk 1 above)
Comment (senjucanon2) [reply 120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java:246]: This was done to be in sync with already implemented GitCaching for pipelines.It was discussed earlier, will get back with the exact reason finalised upon that time.

**Comment by senjucanon2** [comment]
trigger uts, review

**Comment by senjucanon2** [comment]
trigger uts

**Comment by senjucanon2** [comment]
trigger ut7

**Pull request merged**